pkg/kubectl/service.go

```diff
@@ -196,5 +196,8 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 			return nil, fmt.Errorf("unknown session affinity: %s", params["session-affinity"])
 		}
 	}
+	if params["headless"] == "true" {
```

Comment:
At this point the object is built based solely on the parameters found in this map.